src/main/java/net/sf/rails/game/GameManager.java

```diff
@@ -943,12 +955,74 @@ protected boolean save(GameAction saveAction) {
         File file = new File(saveAction.getFilepath());
         try {
             gameSaver.saveGame(file);
-            return true;
         } catch (IOException e) {
             DisplayBuffer.add(this, LocalText.getText("SaveFailed", e.getMessage()));
             log.error("save failed", e);
             return false;
         }
+
+        boolean archive = Config.getBoolean(ARCHIVE_ENABLED, false);
+        if ( archive ) {
+            int count = Config.getInt(ARCHIVE_KEEP_COUNT, 5);
+            if ( count < 1 ) {
+                count = 1;
+            }
+
+            String archiveDir = Config.get(ARCHIVE_DIRECTORY);
+            if ( StringUtils.isBlank(archiveDir) ) {
+                // default to "archive"
+                archiveDir = "archive";
+            }
+            if ( ! archiveDir.startsWith(File.separator) ) {
+                // it should be relative to the current saved files
+                archiveDir = file.getParent() + File.separator + archiveDir;
+            }
+            log.debug("archiving old saved game files to {}", archiveDir);
+
+            File archiveDirFile = new File(archiveDir);
+            if ( ! archiveDirFile.exists() ) {
+                // create it
+                try {
+                    Files.createDirectories(Path.of(archiveDir,File.separator,"dummy"));
+                }
+                catch (IOException e) {
+                    log.warn("Unable to create archive directory {}", archiveDir, e);
+                    archive = false;
+                }
+            } else if ( archiveDirFile.exists() && ! archiveDirFile.isDirectory() ) {
+                log.warn("Archive directory doesn't seem to be a directory?");
+                archive = false;
+            }
+
+            if ( archive ) {
+                // iterate through files in current directory
+                SortedSet<File> files = new TreeSet<>((a, b) -> ComparisonChain.start()
+                        .compare(b.lastModified(), a.lastModified())
+                        .result());
```

Comment:
```suggestion
                SortedSet<File> files = new TreeSet<>(Comparator.comparing(File::lastModified));
```

Reply:
Your suggestion won't work-- if you'll notice in the original code we are comparing `b` to `a` to generate the sorted list in descending order and then pick up our iteration after `count` items (ie the # of items we leave/retain) thereby avoiding an array "location" check on every pass. Your suggestion would sort them in ascending order.

Reply:
So the order needs to be reversed?

Then we could use:

```suggestion
SortedSet<File> files = new TreeSet<>(Comparator.comparing(File::lastModified).reversed());
```

Reply:
Corresponding change should also then be made in `GameSetupController` given this is where this code originated from :)

Either way, I believe from a previous conversation/PR that the general consensus was a preference for `ComparisonChain` vs `Comparator`? So unless there is another compelling reason to make this change I'm going to hold off until this code (and `GameSetupController`) is touched again.

Reply:
As far as I remember we had no consensus yet. The only people discussing were us two right?

The question in my opinion is:
For which functionality do we want to use external libraries?

A lot of functionality that previously required using an external library are now contained directly in Java (for example a comfortable way to create `Comparator` objects). This leads me to the question whether we should still use external libraries for such things? 

About my reason to mark these lines:
I'm only marking this here because you're touching this code. 
This is the best moment where we can think about changing something.
For me it is kind of irrelevant if you copy the code, reformat it or change it in some other way as long as you touch it.

Reply:
I believe @neutronc also weighed in.

> For me it is kind of irrelevant if you copy the code, reformat it or change it in some other way as long as you touch it.

I don't think it is irrelevant though-- if we have multiple ways of doing the same things in the code base, ie picking on this case, ordering a set, then it becomes harder for 1) someone to be able to understand the code (because they keep seeing multiple ways of doing the exact same thing) and 2) know what is the best/preferred/suggested way when they are going to write or change code.

In this specific case, we already have lots of places that are using `ComparisonChain` AND `Comparator` so until we have a decision on which we should be using as a standard/best practice, I don't see any value in picking over over the other. If this was completely new code that would be one thing and hence more debatable but as this was existing logic I brought it over as is.

But you also bring up a good question on third party libraries and frameworks. I am personally a fan of using JDK provided functionality only if it's fully equivalent to what we can get from some of the third party libraries. In most cases, for example, a lot of the Apache Commons libraries (text, lang3, commons-io) provide functionality a heck of a lot more feature rich then the closest JDK alternatives-- just look the string substitution for a good comparison. Often these types of decisions (a vs b) tend to be more of a religious nature type discussion & decision based on an individuals own experience and exposure vs one actually better than the other and so to move forward I think we go with what we have unless there is a compelling reason to switch -- and at which point it should be a task to go though the code based and make that switch-- not to add a new library/whatever and just use for new code but to actually migrate everything.

Reply:
I would like to hear @neutronc opinion about this, so that we find a final answer to this question.

Reply:
Why don't you raise this up on Slack with both examples and pursue the answer there as I believe @blentz mentioned previously not everyone watches here. I don't really care one way or the other.

Reply:
Tough nut, but from this example :  I like the comparator  though i would need to get used to the notation. what i dislike in this example that the explicit Treeset is lacking information about its objects thats confusing for me as non development trained guy :)
So i would vote to go with comparator.